services/alert/migrate_topic_store.go

```diff
@@ -0,0 +1,160 @@
+package alert
+
+import (
+	"fmt"
+
+	"github.com/influxdata/kapacitor/keyvalue"
+	"github.com/influxdata/kapacitor/services/storage"
+	"github.com/pkg/errors"
+)
+
+const (
+	TopicStoreVersionKey = "topic_store_version"
+	TopicStoreVersion2   = "2"
+)
+
+// MigrateTopicStoreV1V2 - Convert a V1 to a V2 topic store.
+// Also ensures that a topic store has a V2 version number set.
+func (s *Service) MigrateTopicStoreV1V2() error {
+	version, err := s.StorageService.Versions().Get(TopicStoreVersionKey)
+	if err != nil && !errors.Is(err, storage.ErrNoKeyExists) {
+		return fmt.Errorf("cannot determine topic store version: %w", err)
+	}
+	if version == TopicStoreVersion2 {
+		s.diag.Info(fmt.Sprintf("Topic Store is already version %s. Cannot upgrade.", TopicStoreVersion2))
```

Comment:
If, after a period of time (a month?), the backup file still exists, it should be deleted here -- @dgnorton

Reply:
Alternately, and more simply, delete the backup if the conversion succeeds. -- @dgnorton